guacamole/src/main/java/org/apache/guacamole/rest/APIRequest.java

```diff
@@ -58,6 +68,18 @@ public APIRequest(HttpServletRequest request,
 
         super(request);
 
+        // Grab the remote host info.
+        if (request.getRemoteHost() != null && !request.getRemoteHost().isEmpty())
```

Comment:
Is it necessary to sanity check `getRemoteHost()` and `getRemoteAddr()` against `isEmpty()`, given that their return values are guaranteed by the standard? Assuming that the empty string, if it ever does occur, is a legitimate value, doesn't this boil down to:

    if (request.getRemoteHost() != null)
        this.remoteHost = request.getRemoteHost();
    else
        this.remoteHost = null;

and thus could simply be:

    this.remoteAddr = request.getRemoteAddr();